structlog/structlog.go

```diff
@@ -0,0 +1,104 @@
+package structlog
```

Comment:
I know this isn't exactly config, but... I'd still recommend keeping it in the `config` package for now.

Motivations being:

1. We have no other use-cases on the radar in the near future.
2. By using `mapstructure`, its usage is basically tied to Viper structs anyway.
3. If we really generalized it enough to be reusable, it should probably be its own repo/project. However, it's tough to see what's "too specialized" in it until you have more use-csaes.

So... IMO, better to keep it in `config` until we have some more use-cases.